live-build/misc/ansible-roles/appliance-build.virtualization-common/tasks/main.yml

```diff
@@ -38,18 +38,94 @@
     group: root
     state: link
 
+#
+# The section below deals with disabling and masking services that need to be
+# disabled on initial install. Some of those services will remain disabled
+# forever, while others could be enabled by the various applications running
+# on the Delphix Appliance.
+#
+# Our approach is to mask services that need to be disabled. Masking a service
+# instead of just disabling it has 2 advantages:
+#  1. A service that is just disabled will not be started automatically by
+#     systemd on boot, however it can be started by another service if it has
+#     a certain set of dependencies on it (such as "Requires", "PartOf"). A
+#     service that is masked cannot be started in any case (although a
+#     service that is already running and has just been masked will not be
+#     stopped automatically).
+#  2. A service that is disabled can get re-enabled by dpkg maintenance scripts
+#     when the package that provides that service is upgraded. A service that
+#     is masked will not be unmasked.
+#
+# While masking a service is sufficient, we also disable most of the services
+# that we mask. This is not strictly necessary, but is done to remain
+# consistent with how we did things in the past. Our current logic that deals
+# with re-enabling those services will both unmask and enable # them.
+#
+# We have divided the disabling and masking of the services in multiple
+# code blocks to group together services that are handled in the same way by
+# the appliance.
+#
+# Note that if you want to modify this list make sure to also update
+# the logic in fix_and_migrate_services() that is invoked during upgrade.
+# You may also want to look at the logic that handles enabling disabling
+# services in dlpx-app-gate.
+#
+
+#
+# The services in this section should always remain disabled & masked.
+#
+- name: Disable services that should never run
+  command: "systemctl disable {{ item }}"
+  with_items:
+    - nginx.service
+    - postgresql.service
+    - systemd-timesyncd.service
+
+- name: Mask services that should never run
+  command: "systemctl mask {{ item }}"
+  with_items:
+    - nginx.service
+    - postgresql.service
+    - systemd-timesyncd.service
```

Comment:
would be good to combine the disable + mask commands into one ansible task so that the list of services isn't duplicated (and is easier to read). E.g.:
```
- name: Disable and mask services that should never run
  shell: |
    systemctl disable {{ item }}
    systemctl mask {{ item }}
  with_items:
    - nginx.service
    - postgresql.service
    - systemd-timesyncd.service
```

Reply:
Better yet, we could probably use the systemd module: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/systemd_module.html

Reply:
Here are the changes using the ansible module. I personally can't say I really like it more than just using the commands, but here are the changes nevertheless: https://github.com/pzakha/appliance-build/commit/92456c12e0bc55ff5cdcf8c9d1f14d886202fd35

This is the output it produces:
```
16:48:15  TASK [appliance-build.virtualization-common : Disable and mask services that should never run] ***
16:48:15  task path: /var/tmp/jenkins/workspace/devops-gate/master/appliance-build-stage1/pre-push/appliance-build/live-build/misc/ansible-roles/appliance-build.virtualization-common/tasks/main.yml:77
16:48:15  changed: [binary] => (item=nginx.service) => {"ansible_loop_var": "item", "changed": true, "enabled": false, "item": "nginx.service", "name": "nginx.service", "status": {}}

16:48:17  changed: [binary] => (item=postgresql.service) => {"ansible_loop_var": "item", "changed": true, "enabled": false, "item": "postgresql.service", "name": "postgresql.service", "status": {}}

16:48:20  changed: [binary] => (item=systemd-timesyncd.service) => {"ansible_loop_var": "item", "changed": true, "enabled": false, "item": "systemd-timesyncd.service", "name": "systemd-timesyncd.service", "status": {}}
16:48:20  
16:48:20  TASK [appliance-build.virtualization-common : Disable and mask services that should not be running by default] ***
16:48:20  task path: /var/tmp/jenkins/workspace/devops-gate/master/appliance-build-stage1/pre-push/appliance-build/live-build/misc/ansible-roles/appliance-build.virtualization-common/tasks/main.yml:91

16:48:23  changed: [binary] => (item=delphix-fluentd.service) => {"ansible_loop_var": "item", "changed": true, "enabled": false, "item": "delphix-fluentd.service", "name": "delphix-fluentd.service", "status": {}}

16:48:25  changed: [binary] => (item=delphix-masking.service) => {"ansible_loop_var": "item", "changed": true, "enabled": false, "item": "delphix-masking.service", "name": "delphix-masking.service", "status": {}}

16:48:28  changed: [binary] => (item=ntp.service) => {"ansible_loop_var": "item", "changed": true, "enabled": false, "item": "ntp.service", "name": "ntp.service", "status": {}}

16:48:31  changed: [binary] => (item=snmpd.service) => {"ansible_loop_var": "item", "changed": true, "enabled": false, "item": "snmpd.service", "name": "snmpd.service", "status": {}}
16:48:31  
16:48:31  TASK [appliance-build.virtualization-common : Mask NFSv3 services that should not be started automatically] ***
16:48:31  task path: /var/tmp/jenkins/workspace/devops-gate/master/appliance-build-stage1/pre-push/appliance-build/live-build/misc/ansible-roles/appliance-build.virtualization-common/tasks/main.yml:108

16:48:34  changed: [binary] => (item=rpcbind.service) => {"ansible_loop_var": "item", "changed": true, "item": "rpcbind.service", "name": "rpcbind.service", "state": "stopped", "status": {}, "warnings": ["Target is a chroot or systemd is offline. This can lead to false positives or prevent the init system tools from working."]}

16:48:36  changed: [binary] => (item=rpcbind.socket) => {"ansible_loop_var": "item", "changed": true, "item": "rpcbind.socket", "name": "rpcbind.socket", "state": "stopped", "status": {}, "warnings": ["Target is a chroot or systemd is offline. This can lead to false positives or prevent the init system tools from working."]}

16:48:38  ok: [binary] => (item=snmpd.service) => {"ansible_loop_var": "item", "changed": false, "item": "snmpd.service", "name": "snmpd.service", "state": "stopped", "status": {}, "warnings": ["Target is a chroot or systemd is offline. This can lead to false positives or prevent the init system tools from working."]}
16:48:38  [WARNING]: Target is a chroot or systemd is offline. This can lead to false
16:48:38  positives or prevent the init system tools from working.
16:48:38  
16:48:38  TASK [appliance-build.virtualization-common : Disable docker] ******************
16:48:38  task path: /var/tmp/jenkins/workspace/devops-gate/master/appliance-build-stage1/pre-push/appliance-build/live-build/misc/ansible-roles/appliance-build.virtualization-common/tasks/main.yml:122

16:48:41  changed: [binary] => (item=docker.service) => {"ansible_loop_var": "item", "changed": true, "enabled": false, "item": "docker.service", "name": "docker.service", "status": {}}
```
Note that I caught a bug where we were masking snmpd twice, which is fixed now. 

The thing I don't like so much is that leaving a service enabled but masked fails with:
```
15:58:43  failed: [binary] (item=rpcbind.service) => {"ansible_loop_var": "item", "changed": false, "item": "rpcbind.service", "msg": "Unable to enable service rpcbind.service: Synchronizing state of rpcbind.service with SysV service script with /lib/systemd/systemd-sysv-install.\nExecuting: /lib/systemd/systemd-sysv-install enable rpcbind\nFailed to enable unit, unit /etc/systemd/system/rpcbind.service is masked.\n"}
```
Instead have to use "state: stopped" with "masked: yes", which makes it different from other calls, and also gives warnings.

With all that said, if both of you prefer the systemd module, I can switch to that.